demo/apisix/conf/config.yaml

```diff
@@ -0,0 +1,4 @@
+deployment:
+  role: data_plane
+  role_data_plane:
+    config_provider: yaml
```

Comment:
Could we try putting the environment variable `APISIX_STAND_ALONE=true` for APISIX container?

I ran APISIX in standalone mode using command
`docker run -it --rm -e APISIX_STAND_ALONE=true apache/apisix:3.2.0-debian` without this conf file.

Reply:
As my comment: https://github.com/linagora/tmail-backend/pull/652#discussion_r1171125022

we can see the `docker-entrypoint.sh`, The `cat` command has been making "empty" file after the mount file

Reply:
Hmmmm actually I gave it a quick look and it is quite confusing... I start to understand why @vttranlina is struggling with it.

Official doc website of APISIX: - https://apisix.apache.org/docs/apisix/deployment-modes/#standalone

Says pretty much add this piece of code in config.yaml to start in standalone mode (what Tung did)

The github repo of the docker apisix project: https://github.com/apache/apisix-docker#how-to-run-apisix-in-stand-alone-mode

Says to set up the env variale like @quantranhong1999 is suggesting. Notice that the link at the end of the section telling you you can know more details about standalone mode towards a doc page on the official website gets you to... a 404 :)

Something clearly not consistent here.

Digging a bit more though: https://github.com/apache/apisix-docker/issues/401

That I believe is the issue you mentionned somewhere @vttranlina ? Notice it has been closed because of inactivity, not because it maybe got solved... Do you still have that issue?

My guess is doc on github repo potentially outdated, and should follow the official one?

Reply:
It seems adding the env variable is overriding the conf files with default standalone conf... from what I understood.

Reply:
> It seems adding the env variable is overriding the conf files with default standalone conf... from what I understood.

Yes, 
This bellow is a shell script of Apisix docker-entrypoint.sh 

```
PREFIX=${APISIX_PREFIX:=/usr/local/apisix}

if [[ "$1" == "docker-start" ]]; then
    if [ "$APISIX_STAND_ALONE" = "true" ]; then
        cat > ${PREFIX}/conf/config.yaml << _EOC_
deployment:
  role: data_plane
  role_data_plane:
    config_provider: yaml
_EOC_

        cat > ${PREFIX}/conf/apisix.yaml << _EOC_
routes:
  -
#END
_EOC_
```

We can see the command 
```
        cat > ${PREFIX}/conf/apisix.yaml << _EOC_
routes:
  -
#END
```
It overrides conf file with "empty routes"

Reply:
Yep saw it as well :)